src/Database/Concerns/DealsWithModels.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Stancl\Tenancy\Database\Concerns;
+
+use Closure;
+use Illuminate\Database\Eloquent\Model;
+use Symfony\Component\Finder\Finder;
+use Symfony\Component\Finder\SplFileInfo;
+
+trait DealsWithModels
+{
+    /**
+     * If this is not null, use this instead of the default model discovery logic.
+     */
+    public static Closure|null $modelDiscoveryOverride = null;
+
+    public static array $modelsExcludedFromDiscovery = [];
+
+    /**
+     * Discover all models in the directories configured in 'tenancy.rls.model_directories'.
+     */
+    public static function getModels(): array
+    {
+        if (static::$modelDiscoveryOverride) {
+            return (static::$modelDiscoveryOverride)();
+        }
+
+        $modelFiles = Finder::create()->files()->name('*.php')->in(config('tenancy.rls.model_directories'));
+
+        // todo1 Add array property for excluding specific models
```

Comment:
Resolve priority todos in this PR

Reply:
Deleted the comment (the `$modelsExcludedFromDiscovery` property was already there, so this comment was a leftover). Seems like this was the only priority to-do added in this PR, so I think we can resolve this.